discovery/gossiper.go

```diff
@@ -1560,10 +1565,16 @@ func (d *AuthenticatedGossiper) networkHandler() {
 		// have been dropped, or not properly propagated through the
 		// network.
 		case tick := <-d.cfg.RetransmitTicker.Ticks():
-			if err := d.retransmitStaleAnns(tick); err != nil {
-				log.Errorf("unable to rebroadcast stale "+
-					"announcements: %v", err)
-			}
+			d.wg.Add(1)
+			go func() {
+				defer d.wg.Done()
+
+				err := d.retransmitStaleAnns(tick)
```

Comment:
Looks good to me, as far as I can tell `retransmitStaleAnns` doesn't access anything that could cause a concurrent access issue with the main goroutine.

Reply:
Isn't `SendMessageLazy` already async? This is what's used inside the `Broadcast`,
https://github.com/lightningnetwork/lnd/blob/8044891df418234e6564060cc5776670b4bd614d/discovery/gossiper.go#L1812-L1816

Reply:
Good observation, seems like we would also need to add some kind of deduplicate logic in the `BroadcastMessage` as well ?

Reply:
I also think this can even happen today without doing the `retransmitStaleAnns ` async, because the rpcCall to update the chanPolicy does not go through the `ProcessRemoteAnnouncement` loop.

https://github.com/lightningnetwork/lnd/blob/1db6c31e2084fbacea443880d61830dba130c99a/discovery/gossiper.go#L624-L640